src/Autocomplete/src/AutocompleteResults.php

```diff
@@ -14,11 +14,25 @@
 final class AutocompleteResults
 {
     /**
-     * @param list<array{label: string, value: mixed}> $results
+     * @param list<array{text: string, value: mixed, group_by?: string}> $options
+     * @param list<array{label: string, value: string}>                  $optgroups
      */
     public function __construct(
-        public array $results,
+        public array $options,
+        public array $optgroups,
         public bool $hasNextPage,
```

Comment:
We can't break BC here, even though it's unlikely to cause problems. So:

A) Rename `$options` back to `$results`
B) Move `$optgroups` to the final argument and make it optional

Then, let's move the `getResults()` logic into the controller. That'll keep this class dead-simple - just a carrier of data.